# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/DMABufSurface.h
# Commit: 77959a8de7c3
# Full Hash: 77959a8de7c364b731e4da006033d9807cc12310
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2025-04-29 20:49:47
# Regressor Bug: 1959509
# File Overlap Count: 1
# Description:
#   Bug 1959509 [Linux] Create DMABuf YUV planes with flags and modifiers to allow direct rendering r=emilio
#   
#   Use DMABuf modifiers from DRMFormat and GBM flags to create DMABuf with propper params for direct scanout.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D245885
# ==============================================================================

diff -r 1c5f8d98e479 -r 77959a8de7c3 widget/gtk/DMABufSurface.h
--- a/widget/gtk/DMABufSurface.h	Tue Apr 29 07:15:44 2025 +0000
+++ b/widget/gtk/DMABufSurface.h	Tue Apr 29 07:17:18 2025 +0000
@@ -261,6 +261,7 @@
   int32_t mOffsets[DMABUF_BUFFER_PLANES];
 
   struct gbm_bo* mGbmBufferObject[DMABUF_BUFFER_PLANES];
+  uint32_t mGbmBufferFlags;
 
 #ifdef DEBUG
   void* mMappedRegion[DMABUF_BUFFER_PLANES];
@@ -366,7 +367,6 @@
 
   EGLImageKHR mEGLImage;
   GLuint mTexture;
-  uint32_t mGbmBufferFlags;
   uint64_t mBufferModifier;
 };
 
@@ -446,8 +446,10 @@
   ~DMABufSurfaceYUV();
 
   bool Create(const mozilla::layers::SurfaceDescriptor& aDesc) override;
-  bool CreateYUVPlane(mozilla::gl::GLContext* aGLContext, int aPlane);
-  bool CreateYUVPlaneGBM(int aPlane);
+  bool CreateYUVPlane(mozilla::gl::GLContext* aGLContext, int aPlane,
+                      mozilla::widget::DRMFormat* aFormat = nullptr);
+  bool CreateYUVPlaneGBM(int aPlane,
+                         mozilla::widget::DRMFormat* aFormat = nullptr);
   bool CreateYUVPlaneExport(mozilla::gl::GLContext* aGLContext, int aPlane);
 
   bool MoveYUVDataImpl(const VADRMPRIMESurfaceDescriptor& aDesc, int aWidth,
